### PR TITLE
feat: add aspect ratio for ocr predictor

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR adds the option to preserve aspect ratio of the pages in the whole OCRPredictor pipeline, which leads to much better results on the detection task. 
Any feedback is welcome!

### CODE REVIEW

File: doctr/models/predictor/base.py [hunk 1]
```diff
@@ -34,12 +34,16 @@ def __init__(
         self,
         assume_straight_pages: bool = True,
         straighten_pages: bool = False,
+        preserve_aspect_ratio: bool = True,
+        symmetric_pad: bool = True,
```
Comment (fg-mindee) [inline doctr/models/predictor/base.py:38]: The docstring needs to be updated :)

**Review by fg-mindee — CHANGES_REQUESTED** [review]
Thanks for the PR!
I added a few comments, could you confirm whether this improves perf with the existing checkpoints? (it isn't specified in the PR details, so I prefer to make sure we use everything)

If that improves for existing checkpoints (trained with images that were stretched, that should strongly indicate that we should update our resizing strategy for detection training asap :))

**Comment by charlesmindee** [comment]
Yes @fg-mindee it doesn't hurt performances for now, and I will retrain all detection models preserving the aspect ratio to have a coherent pipeline, because it is way better not to deform writtings on documents for the model convergence!

**Review by fg-mindee — DISMISSED** [review]
Nice!

**Review by fg-mindee — APPROVED** [review]
Thanks!

**Pull request merged**